src/inet/TCPEndPointImplSockets.cpp

```diff
@@ -127,6 +127,7 @@ CHIP_ERROR TCPEndPointImplSockets::BindImpl(IPAddressType addrType, const IPAddr
 
         if (res == CHIP_NO_ERROR)
         {
+            VerifyOrReturnValue(mSocket >= 0, CHIP_ERROR_INCORRECT_STATE);
```

Comment:
Why is this being checked here but not above in the `reuseAddr` case ?   Either mSocket has to be valid here, or we should be checking it everywhere, no?

But also, GetSocket() returned success, so I would think it has to be valid here.